### PR TITLE
Filter Out Invalid Uniswap V2 Pools

### PR DESCRIPTION
**Pull request by nlordell**
This PR fixes an issue where the `driver` was sending invalid UniV2 pools to the solver engine. The engine expects ~~`uint96`~~ `uint112` reserves, but it is possible to set up a pools that has more reserves than that (effectively bricking it).

### Test Plan

Ran locally on Goerli and swapped with WETH <> `0xdc31ee1784292379fbb2964b3b9c4124d8f89c60` pool.

### CODE REVIEW

File: crates/driver/src/boundary/liquidity/uniswap/v2.rs [hunk 1]
```diff
@@ -43,7 +43,14 @@ pub fn to_domain(id: liquidity::Id, pool: ConstantProductOrder) -> liquidity::Li
         .downcast_ref::<uniswap_v2::Inner>()
         .expect("downcast uniswap settlement handler");
 
-    liquidity::Liquidity {
+    // Trading on Uniswap V2 pools where the reserves overflows `uint96`s does
+    // not work, so filter these pools out.
+    let max = (1_u128 << 96) - 1;
```
Comment (vkgnosis) [inline crates/driver/src/boundary/liquidity/uniswap/v2.rs:48]: Slight preference to using `pow` rather than `<<`.

**Review by vkgnosis — APPROVED** [review]
Do we not have the same problem in the pre colocation code?

**Comment by nlordell** [comment]
> Do we not have the same problem in the pre colocation code?

No, we filter this out at a different stage:

https://github.com/cowprotocol/services/blob/f73f0ddb8a2fad22f6b6d1b19226eeb1ca6b9d86/crates/shared/src/sources/uniswap_v2/pool_fetching.rs#L24C16-L24C33

Although, I realize the bit width I used was wrong.

File: crates/driver/src/boundary/liquidity/uniswap/v2.rs (hunk 1 above)
Comment (nlordell) [reply crates/driver/src/boundary/liquidity/uniswap/v2.rs]: Changed.

**Review by MartinquaXD — APPROVED** [review]

**Pull request merged**